src/VisualStudio/Core/Def/Implementation/Log/VisualStudioErrorLogger.cs

```diff
@@ -4,23 +4,51 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.ErrorLogger;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Watson;
 using Microsoft.VisualStudio.Shell;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.Log
 {
-    [ExportWorkspaceService(typeof(IErrorLoggerService), ServiceLayer.Host),  Export(typeof(IErrorLoggerService)), Shared]
+    [ExportWorkspaceService(typeof(IErrorLoggerService), ServiceLayer.Host), Export(typeof(IErrorLoggerService)), Shared]
     internal class VisualStudioErrorLogger : IErrorLoggerService
     {
-        public void LogError(string source, string message)
+        public void LogException(object source, Exception exception)
         {
-            ActivityLog.LogError(source, message);
+            var name = source.GetType().Name;
+            ActivityLog.LogError(name, ToLogFormat(exception));
+
+            if (source is ReportCrashDumpsToMicrosoft)
```

Comment:
Why not just check to see if the name of the crashing code fix / refactoring provider appears in ```PredefinedCodeFixProviderNames``` or ```PredefinedCodeRefactoringProviderNames```? That way, we won't need to remember to inherit the interface for any new fix / refactoring providers we add. All we would need to do is have its name be in one of the predefined lists...